tgui/packages/tgui/interfaces/Manufacturer/components/BlueprintButton.tsx

```diff
@@ -0,0 +1,181 @@
+import { CenteredText } from './CenteredText';
+import { truncate } from '../../../format';
+import { BlueprintButtonStyle, BlueprintMiniButtonStyle } from '../constant';
+
+const getBlueprintTime = (time, manufacturerSpeed) => {
+  return round(time / 10 / manufacturerSpeed, 0.01);
+};
+
+/*
+Get whether or not there is a sufficient amount to make. does NOT affect sanity checks on the DM side,
+this just offloads some of the computation to the client.
+
+Consequently, if the checks on the .dm end change, this has to change as well.
+
+Only the DM checks matter for actually making the item though, this just enables and disables buttons /
+shows what materials are missing.
+*/
+const getProductionSatisfaction = (
+  pattern_requirements:string[],
+  amount_requirements:number[],
+  materials_stored:ResourceData[]) =>
+{
+  // Copy values of mats stored to edit in case we need to try the same material twice
+  let material_amts_predicted:Record<string, number> = {};
+  materials_stored.forEach((value:ResourceData) => (
+    material_amts_predicted[value.id] = value.amount
+  ));
+  let patterns_satisfied:boolean[] = [];
+  for (let i in pattern_requirements) {
+    const target_pattern = pattern_requirements[i];
+    const target_amount = amount_requirements[i];
+    const matchingMaterial = materials_stored.find((material:ResourceData) => (
+      target_pattern === "ALL" || material.satisfies.find((pattern:string) => (pattern === target_pattern)) !== undefined
+    ));
+    if (matchingMaterial !== undefined && matchingMaterial.amount >= target_amount/10) {
+      material_amts_predicted[i] -= target_amount/10;
+      patterns_satisfied.push(true);
+    }
+    else {
+      patterns_satisfied.push(false);
+    }
+  }
+  return patterns_satisfied;
+};
+
+export const BlueprintButton = (props:BlueprintButtonData) => {
+
+  const {
+    actionVendProduct,
+    blueprintData,
+    materialData,
+    manufacturerSpeed,
+  } = props;
+  const blueprintSatisfaction = getProductionSatisfaction(
+    blueprintData.item_paths,
+    blueprintData.item_amounts,
+    materialData,
+  );
+  // Condense producability
+  const notProduceable = blueprintSatisfaction.some((materialIsProducable:boolean) => materialIsProducable === false);
+  // Don't include this flavor if we only output one item, because if so, then we know what we're making
+  const outputs = (blueprintData.item_names.length < 2
+    && blueprintData.create < 2
+    && !blueprintData.apply_material) ? null : (
+      <>
+        <br />
+        Outputs: <br />
+        {blueprintData.item_names.map((value:string, index:number) => (
+          <b key={index}>
+            {blueprintData.create}x {value}<br />
+          </b>
+        ))}
+      </>
+    );
+  const content_requirements = (
+    <Section
+      title="Requirements"
+      buttons={<Button icon="hourglass" backgroundColor="rgba(0,0,0,0)">{getBlueprintTime(blueprintData.time, manufacturerSpeed)}s</Button>}
```

Comment:
https://discord.com/channels/182249960895545344/890226559691157524/1230160837365202995
I was not able to find any other way to place an element to the right-hand side so that the time indicator could take up less vertical space on the tooltip.
Understandably this is not ideal so if there is a way with sections to not *have* to only place buttons on the right hand side, and rather other elements, or if something else should be done for now entirely, that's fine

Reply:
Gotcha. Fine for now (I mean, eh, but I'm not going to require you to go fix the thing upstream then wait for all that). Broadly, I'm gonna have to go huff at the tgstation folk about their Section component but also we're not on React yet so yaaay.